File: server/brownshome/scriptwars/game/tanks/RandomAI.java
package brownshome.scriptwars.game.tanks;

import java.io.IOException;
import java.util.Scanner;

public class RandomAI {
	/**
	 * The main method to start the AI and connect to the server.
	 * 
	 * args[0] should contain the game id.
	 * You can request one from: http://script-wars.com/games/Tanks
	 * by clicking the 'Join' button.
	 * 
	 * @param args The input arguments containing the ID allocated by the server
	 * @throws IOException If we failed to connect to the server
	 */
	public static void main(String[] args) throws IOException {
		// args[0] should contain the game id.
		// You can request one from: http://script-wars.com/games/Tanks
		// by clicking the 'Join' button
		
		int id;
		if(args.length > 0){
			id = Integer.valueOf(args[0]);
		} else {
			//System.out.println("Usage: JAVACOMMAND serverid");
			//System.exit(1);
			return;
		}

		TankAPI api = new TankAPI(id, "www.script-wars.com", "Random AI");

		while(api.nextTick()) {
			if(!api.isAlive()) {
				continue;
			}

			// Move randomly, this will be overwritten if we can see someone.
			int direction = (int) (Math.random() * 4);
			api.move(Direction.values()[direction]);
			
			// See if there is a tank in our field of view,
			// and if there is select it.
			Tank targetTank = null;
			for(Tank tank : api.getVisibleTanks()){
				targetTank = tank;
			}
			
			// If we can see a tank, lets shoot it.
			if(targetTank != null){
				Coordinates targetPosition = targetTank.getPosition();
				Coordinates myPosition = api.me().getPosition();
				
				Direction targetDirection = Direction.getDirection(targetPosition, myPosition);
				if(targetDirection != null) {
					//We have a clear shot on the target
					api.shoot(targetDirection);
				}
			}
			
			//System.out.println("Position: " + api.me().getPosition());
		}
		
		//System.out.println("Disconnected from server:\n\t" + api.getConnectionStatus());
	}
}

File: client/csharp/Script Wars Client/Script Wars AI/ScriptWars/Games/Tanks/PlayerBots/RandomAi.cs
using System;

namespace ScriptWars.Games.Tanks.PlayerBots
{
    /// <summary>
    /// Example Script Wars tanks AI using the C# client.
    /// </summary>
    public class RandomAi
    {
        public static void Main(string[] args)
        {
            int id;

            // Try parse ID from args
            if (!(args.Length > 0 && int.TryParse(args[0], out id)))
            {
                // Try get ID from console input
                bool parsed;

                do
                {
                    Console.WriteLine("Enter ID: ");
                    parsed = int.TryParse(Console.ReadLine(), out id);
                } while (!parsed);
            }

            var ai = new RandomAi(id);
            ai.Run();
        }
        
        private readonly TankApi _api;
        private readonly Random _random = new Random();

        private RandomAi(int id)
        {
            _api = new TankApi(id, "www.script-wars.com", "C# Random AI");
        }

        private void Run()
        {
            var directionCount = Enum.GetValues(typeof(Direction)).Length;
            
            while (_api.NextTick())
            {
                if (!_api.IsAlive)
                {
                    // Dead this tick
                    continue;
                }

                // Pick a random non-wall and non-tank direction to move in
                Direction direction;
                Coordinates newPosition;
                do
                {
                    direction = (Direction) _random.Next(directionCount);
                    newPosition = direction.Move(_api.Position);
                } while (_api.Map.IsWall(newPosition) || _api.Map.GetTank(newPosition) != null);

                _api.Move(direction);

                // Try shoot at someone if there's ammo
                if (_api.Ammo > 0)
                {
                    foreach (var tank in _api.VisibleTanks)
                    {
                        var targetDirection = DirectionExtensions.GetDirection(_api.Position, tank.Position);

                        // Check for cardinal direction shot, shoot if we can
                        if (targetDirection.HasValue)
                        {
                            _api.Shoot(targetDirection.Value);
                        }
                    }
                }

                _api.PrintAction();
            }
        }
    }
}